### PR TITLE
Data persistence

### LINKED ISSUE #249: refreshtoken/accesstoken persistence when refreshing
We need a way to persist user object and refresh token so that we won't be taken back to the login page on every refresh. The most optimal way to do this in my opinion would be to use localStorage. We could take inspiration from [this video](https://youtu.be/27KeYk-5vJw).

### PR DESCRIPTION
**Pull request by NielsPraet**
**New feature checklist**:
- [x] I've thoroughly documented the code

This PR adds a component that makes it possible to restore previous recent sessions so that you won't get kicked back to the login when refreshing the page.

Closes #249

### CODE REVIEW

File: frontend/components/PersistLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          }
+
+          if (user) {
+            setUser(JSON.parse(user));
+          }
+        }
+
+        await refresh();
+      } catch (err) {
+        console.log(err);
```
Comment (Mikxox) [inline frontend/components/PersistLogin.tsx:58]: Probably forgot to remove this console log

File: frontend/hooks/useRefreshToken.ts [hunk 2]
```diff
@@ -26,9 +33,15 @@ const useRefreshToken = () => {
...
+    const refreshToken = response.data.refreshToken;
+
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('refreshToken', refreshToken);
+    }
+
     setTokens({
       accessToken: response.data.accessToken,
-      refreshToken: response.data.refreshToken,
+      refreshToken,
```
Comment (Mikxox) [inline frontend/hooks/useRefreshToken.ts:44]: Can you change this to refreshToken: refreshToken for consistency?

**Review by Mikxox — APPROVED** [review]

**Comment by Mikxox** [comment]
Seems to be working as intended 👍  very easy to use on other pages as well.
I'm approving this with the idea that besides the header and login page no other pages got changed but it is really hard to tell with how github is showing the diffs.

**Review by NinjaJay — APPROVED** [review]

File: frontend/hooks/useRefreshToken.ts (hunk 2 above)
Comment (NielsPraet) [reply frontend/hooks/useRefreshToken.ts:44]: Well.... ES6 has a nifty way to deal with that redundant information... :(

**Comment by NielsPraet** [comment]
@Mikxox Can you take a look at this merge conflict because I'm having a hard time wrapping my head around the JSX code

**Comment by Mikxox** [comment]
> @Mikxox Can you take a look at this merge conflict because I'm having a hard time wrapping my head around the JSX code

Should be fixed now, github has some real trouble with html diffs it seems.

**Pull request merged**